packages/medusa-plugin-auth/src/core/validate-callback.ts

```diff
@@ -1,7 +1,38 @@
 import { CustomerService, UserService } from '@medusajs/medusa';
 import { MedusaError } from 'medusa-core-utils';
 import { EntityManager } from 'typeorm';
-import { CUSTOMER_METADATA_KEY } from '../types';
+import { CUSTOMER_METADATA_KEY, AUTH_PROVIDER_KEY, AuthOptions } from '../types';
+import { AUTH0_ADMIN_STRATEGY_NAME, AUTH0_STORE_STRATEGY_NAME } from '../auth-strategies/auth0/types';
+import { FACEBOOK_ADMIN_STRATEGY_NAME, FACEBOOK_STORE_STRATEGY_NAME } from '../auth-strategies/facebook/types';
+import { GOOGLE_ADMIN_STRATEGY_NAME, GOOGLE_STORE_STRATEGY_NAME } from '../auth-strategies/google/types';
+import { LINKEDIN_ADMIN_STRATEGY_NAME, LINKEDIN_STORE_STRATEGY_NAME } from '../auth-strategies/linkedin/types';
+
+type StrategyErrorIdentifierType = keyof AuthOptions;
+type StrategyNames = {
+	[key in StrategyErrorIdentifierType]: {
+		admin: string;
+		store: string;
+	};
+};
+
+const strategyNames: StrategyNames = {
```

Comment:
todo: Move that into the strategies directory in the types. I don't think the core should hold it. plus it make it more complicated for a user to add a new strategy without forgetting that place. wdyt?

Reply:
I totally agree